packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -23,20 +32,56 @@ export enum SearchIndexesStatuses {
 type SearchIndexesStatus = keyof typeof SearchIndexesStatuses;
 
 export enum ActionTypes {
-  SetStatus = 'indexes/regular-indexes/SetStatus',
+  SetStatus = 'indexes/search-indexes/SetStatus',
+  OpenCreateSearchIndexModal = 'indexes/search-indexes/OpenCreateSearchIndexModal',
+  CloseCrudSearchIndexModal = 'indexes/search-indexes/CloseCrudSearchIndexModal',
+  SetError = 'indexes/search-indexes/SearchError',
+  SetBusy = 'indexes/search-indexes/SetBusy',
 }
 
 type SetStatusAction = {
   type: ActionTypes.SetStatus;
   status: SearchIndexesStatus;
 };
 
+type OpenCreateSearchIndexModalAction = {
+  type: ActionTypes.OpenCreateSearchIndexModal;
+};
+
+type CloseCrudSearchIndexModalAction = {
+  type: ActionTypes.CloseCrudSearchIndexModal;
+};
+
+type SetErrorAction = {
+  type: ActionTypes.SetError;
+  error: string | undefined;
+};
+
+type SetBusyAction = {
+  type: ActionTypes.SetBusy;
+  busy: boolean;
+};
```

Comment:
Setter actions are an antipattern (for a good reason, you need to read all the action code to understand what setBusy does), based on the code those should be search index creation start,  success, and error actions. Here's some relevant things for how to structure this better

https://redux.js.org/style-guide/#model-actions-as-events-not-setters
https://redux.js.org/style-guide/#write-meaningful-action-names
https://redux.js.org/style-guide/#avoid-dispatching-many-actions-sequentially

Reply:
Yeah you are right. We were just @mabaasit and I discussing how to improve this, we changed the actions to start, failed and succeeded. I'll push the changes when I fix the tests.